tests/test_wait.py

```diff
@@ -0,0 +1,73 @@
+from datetime import timedelta
+
+import requests
+from pytest import approx, raises
+
+from pytest_httpserver import HTTPServer
+
+
+def test_wait_success(httpserver: HTTPServer):
+    waiting_timeout = 0.1
+
+    with httpserver.wait(stop_on_nohandler=False, timeout=waiting_timeout) as waiting:
+        requests.get(httpserver.url_for("/foobar"))
+        httpserver.expect_oneshot_request("/foobar").respond_with_data("OK foobar")
+        requests.get(httpserver.url_for("/foobar"))
+    assert waiting.result
+
+    httpserver.expect_oneshot_request("/foobar").respond_with_data("OK foobar")
+    httpserver.expect_oneshot_request("/foobaz").respond_with_data("OK foobaz")
+    with httpserver.wait(timeout=waiting_timeout) as waiting:
+        requests.get(httpserver.url_for("/foobar"))
+        requests.get(httpserver.url_for("/foobaz"))
+    assert waiting.result
+
+
+def test_wait_unexpected_request(httpserver: HTTPServer):
+    def make_unexpected_request_and_wait() -> None:
+        with raises(AssertionError) as error:
+            waiting_timeout = 0.1
+            with httpserver.wait(raise_assertions=True, stop_on_nohandler=True, timeout=waiting_timeout) as waiting:
+                requests.get(httpserver.url_for("/foobaz"))
+            assert not waiting.result
+        no_handler_text = 'No handler found for request'
+        assert no_handler_text in str(error)
+
+    make_unexpected_request_and_wait()
+
+    httpserver.expect_oneshot_request("/foobar", ordered=True).respond_with_data("OK foobar")
+    httpserver.expect_oneshot_request("/foobaz", ordered=True).respond_with_data("OK foobaz")
+    make_unexpected_request_and_wait()
+
+
+def test_wait_timeout(httpserver: HTTPServer):
+    httpserver.expect_oneshot_request("/foobar").respond_with_data("OK foobar")
+    httpserver.expect_oneshot_request("/foobaz").respond_with_data("OK foobaz")
+    waiting_timeout = 1
+    with raises(AssertionError) as error:
+        with httpserver.wait(raise_assertions=True, timeout=waiting_timeout) as waiting:
+            requests.get(httpserver.url_for("/foobar"))
+        assert not waiting.result
+        waiting_time_error = 0.1
+        assert waiting.time.seconds == approx(waiting_timeout, abs=waiting_time_error)
```

Comment:
Do we need approx here? Wouldn't be a `>` enough?

```
waiting.time.seconds >= 1
```
Have you seen any test run where it was less than 1?

Reply:
There is a chance that the timer will timeout sooner than in 1 second. See https://stackoverflow.com/a/1133879/2108548

Reply:
Thanks for the clarification.
Why I'm worring about is that whether a weaker (or busier) hardware would be incapable of doing the things in that 0.1 seconds threshold, I think that this would potentially add flakiness to this test.
But let's go with it for now, and we will see what travis and others will say about it. Maybe we could increase the threshold later if needed.